packages/playwright/src/reporters/junit.ts

```diff
@@ -79,16 +79,24 @@ class JUnitReporter extends EmptyReporter {
 
     serializeXML(root, tokens, this.stripANSIControlSequences);
     const reportString = tokens.join('\n');
-    if (this.outputFile) {
-      assert(this.config.configFile || path.isAbsolute(this.outputFile), 'Expected fully resolved path if not using config file.');
-      const outputFile = this.config.configFile ? path.resolve(path.dirname(this.config.configFile), this.outputFile) : this.outputFile;
+    const outputFile = this.resolveOutputFile();
+    if (outputFile) {
       await fs.promises.mkdir(path.dirname(outputFile), { recursive: true });
       await fs.promises.writeFile(outputFile, reportString);
     } else {
       console.log(reportString);
     }
   }
 
+  private resolveOutputFile(): string | undefined {
```

Comment:
I extracted this method to avoid duplication when resolving the outputFile when resolving attachment paths.